### PR TITLE
Fix CLI

### PR DESCRIPTION
**Pull request by fmvilas**
The CLI is not working. I fixed it and hope it's correct. Let me know if you want me to do something else before merging.

### CODE REVIEW

**Comment by fmvilas** [comment]
@derberg it looks like you added the glob package for Windows compatibility but it's broken for me on Mac. Any reason you used it?

**Comment by derberg** [comment]
@fmvilas here you have all the details https://github.com/Mermade/mdv/pull/43

**Comment by fmvilas** [comment]
Ok, I see. Let me see how can I get it to work using glob so it doesn't fail on mac either.

**Comment by fmvilas** [comment]
Found the problem. It's calling an asynchronous function (`glob`) inside a synchronous `for` loop. I opted for using the sync version of glob since it's a CLI and we'll not notice much performance degradation.

File: mdv.js [hunk 1]
```diff
@@ -34,45 +34,43 @@ const argv = require('yargs')
...
 const options = argv;
 
-
 for (let a of argv._) {
-    glob(a, options, function (er, files) {
-        for (let file of files) {
-            const s = fs.readFileSync(a,'utf8');
-            options.source = file;
-            const result = validator.validate(s,options);
+    const files = glob.sync(a, { debug: !!options.debug });
```
Comment (fmvilas) [inline mdv.js:38]: We were passing `options` before but the only `flag` that this CLI and `glob` have in common, so I simplified it.

**Comment by derberg** [comment]
I'm wondering why the tests were passing, did you check if they are passing now after actually checking all works fine?

File: mdv.js [hunk 2]
```diff
@@ -34,45 +34,43 @@ const argv = require('yargs')
...
+    for (let file of files) {
+        const s = fs.readFileSync(a,'utf8');
```
Comment (MikeRalphson) [inline mdv.js:40]: I think this `readFileSync` should be of `file` not `a`.

File: mdv.js [hunk 3]
```diff
@@ -34,45 +34,43 @@ const argv = require('yargs')
...
+        options.source = file;
+        const result = validator.validate(s,options);
 
-            if (options.save) {
-                fs.writeFileSync(a+'.html',options.html,'utf8');
-                delete options.html;
-            }
+        if (options.save) {
+            fs.writeFileSync(a+'.html',options.html,'utf8');
```
Comment (MikeRalphson) [inline mdv.js:45]: Maybe here also?

**Review by MikeRalphson — CHANGES_REQUESTED** [review]
I was testing with `mdv *.md` and `mdv '*.md'` and it still wasn't right.

File: mdv.js (hunk 2 above)
Comment (fmvilas) [reply mdv.js]: Good catch! I had it on my first iteration and lost it for some reason.

File: mdv.js (hunk 3 above)
Comment (fmvilas) [reply mdv.js]: Good point. Fixed.

**Comment by fmvilas** [comment]
Review suggestions applied and tests are passing :)

**Review by MikeRalphson — APPROVED** [review]
LGTM now, thanks.

**Comment by fmvilas** [comment]
In any case, tests need to be reviewed too because I just tried and they were passing even on the current version.

**Comment by MikeRalphson** [comment]
OK, will revisit the tests. Do you want a release published now or to wait for that?

**Comment by fmvilas** [comment]
I'd release this one. At least it's already better than it was. Looking at tests now but my brain is fried so no promises 😄

**Comment by MikeRalphson** [comment]
The tests bypass the CLI and call the API directly so I think that's why.

**Comment by MikeRalphson** [comment]
@fmvilas @derberg `mdv@1.3.2` published with the above fixes. Thanks both!

**Comment by fmvilas** [comment]
Yeah, that's why. I can see it's failing successfully on tests.

PS: Love that term 😄 "failing successfully"

**Pull request merged**